### PR TITLE
bulk ev4

### PR DESCRIPTION
**Pull request by benauriemma**

### CODE REVIEW

File: rackcity/views/it_model_views.py [hunk 1]
```diff
@@ -539,21 +542,9 @@ def model_bulk_export(request):
...
-                "failure_message": Status.EXPORT_ERROR.value
-                + GenericFailure.FILTER.value,
-                "errors": str(error),
-            },
-            status=HTTPStatus.BAD_REQUEST,
-        )
-    for filter_arg in filter_args:
-        models_query = models_query.filter(**filter_arg)
-
+    filtered_query, failure_response = get_filtered_query(models_query, request.data)
```
Comment (juliasaveliff) [inline rackcity/views/it_model_views.py:545]: noice

File: rackcity/api/serializers/asset_serializers.py [hunk 2]
```diff
@@ -157,7 +157,7 @@ class RecursiveAssetSerializer(serializers.ModelSerializer):
     chassis_slot = RCIntegerField(
         allow_null=True, max_value=2147483647, min_value=0, required=False
     )
-    offline_storage_site = SiteSerializer()
+    offline_storage_site = serializers.SerializerMethodField()
```
Comment (juliasaveliff) [inline rackcity/api/serializers/asset_serializers.py:160]: claire has also been makin asset serializers changes so watch out for merge conflicts

File: rackcity/utils/rackcity_utils.py [hunk 3]
```diff
@@ -275,24 +275,54 @@ def empty_vs_null_comparison(existing_value, new_value):
...
-            location_occupied_by[rack] = {}
-        for location in asset_location_range:
-            if location in location_occupied_by[rack]:
+        if asset_data["offline_storage_site"]:
+            continue
+        model = ITModel.objects.get(id=asset_data["model"])
+        if model.is_rackmount():
+            rack = asset_data["rack"]
+            height = model.height
+            rack_position = int(asset_data["rack_position"])
```
Comment (juliasaveliff) [inline rackcity/utils/rackcity_utils.py:288]: are these values guaranteed to exist at this point in the codepath?

File: rackcity/utils/rackcity_utils.py [hunk 4]
```diff
@@ -275,24 +275,54 @@ def empty_vs_null_comparison(existing_value, new_value):
...
+                        asset_name = asset_data["hostname"]
+                    else:
+                        asset_name = "unnamed_asset_" + str(unnamed_asset_count)
+                        unnamed_asset_count += 1
+                    rack_location_occupied_by[rack][location] = asset_name
+        else:
+            if "chassis" in asset_data:
+                chassis = asset_data["chassis"]
+            else:
+                chassis = "new_" + asset_data["chassis_number"]
```
Comment (juliasaveliff) [inline rackcity/utils/rackcity_utils.py:316]: tricky

File: rackcity/utils/rackcity_utils.py (hunk 3 above)
Comment (benauriemma) [reply rackcity/utils/rackcity_utils.py:288]: Iff the file is error free. Error handling in bulk is not amazing

**Pull request merged**